pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -171,68 +160,83 @@ public synchronized boolean containsFunction(String tenant, String namespace, St
         return containsFunctionMetaData(tenant, namespace, functionName);
     }
 
-    /**
-     * Sends an update request to the FMT (Function Metadata Topic)
-     * @param functionMetaData The function metadata that needs to be updated
-     * @return a completable future of when the update has been applied
-     */
-    public synchronized CompletableFuture<RequestResult> updateFunction(FunctionMetaData functionMetaData) {
-
-        FunctionMetaData existingFunctionMetadata = null;
-        if (containsFunction(functionMetaData.getFunctionDetails().getTenant(),
-                functionMetaData.getFunctionDetails().getNamespace(),
-                functionMetaData.getFunctionDetails().getName())) {
-            existingFunctionMetadata = getFunctionMetaData(functionMetaData.getFunctionDetails().getTenant(),
-                    functionMetaData.getFunctionDetails().getNamespace(),
-                    functionMetaData.getFunctionDetails().getName());
+    public synchronized void updateFunctionOnLeader(FunctionMetaData functionMetaData, boolean delete)
```

Comment:
Can you add some comments to this method as well of the FunctionMetadataManager to explain the workflow of processing a metadata request?

Reply:
Added more comments on this class